File: Python3/Binary Search/374. Guess Number Higher or Lower.py
""" 374. Guess Number Higher or Lower
Difficulty: Me
Submission Link: https://leetcode.com/problems/guess-number-higher-or-lower/submissions/1376858203/ """

class Solution:
    def guessNumber(self, n: int) -> int:
        left, right = 1, n
        while left <= right:
            mid = (left + right) // 2
            result = guess(mid)  # type: ignore

            if result == 0:
                return mid
            elif result > 0:
                left = mid + 1
            else:
                right = mid - 1

        return -1

File: C++/Binary Search/374. Guess Number Higher or Lower.cpp
/* 374. Guess Number Higher or Lower
Difficulty: Me
Submission Link: https://leetcode.com/problems/guess-number-higher-or-lower/submissions/1376858616/ */

class Solution {
public:
    int guessNumber(int n) {
        
            int start = 1;
            int end = n;
            while(start <= end){
                int mid = start + (end-start) / 2;
            
                if(guess(mid) == -1){
                    end = mid;
                }
                else if(guess(mid) == 1){
                    start = mid+1;
                }
                else{
                    return mid;
                }
        }
        return 0;
    }
};